docker/run_on_docker.sh

```diff
@@ -85,23 +85,34 @@ bootstrap_image() {
   rm -fR $build_workdir
 }
 
+push_image() {
+    local image_name="$1"
+    docker push $image_name || die "Failed to push docker image $image_name"
```

Comment:
Maybe warning instead of failure shoud be the default behavior. Do you think that this should be configurable, through a flag or environment variable for example?